src/schneider_lab_to_nwb/schneider_2024/schneider_2024_intrinsic_signal_imaging_interface.py

```diff
@@ -0,0 +1,91 @@
+from pynwb.base import Images
+from pynwb.image import GrayscaleImage, RGBImage
+from pydantic import DirectoryPath
+import numpy as np
+from PIL import Image
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import DeepDict
+from neuroconv.tools import nwb_helpers
+
+
+class Schneider2024IntrinsicSignalOpticalImagingInterface(BaseDataInterface):
+    """Intrinsic signal optical imaging interface for schneider_2024 conversion"""
+
+    keywords = ["intrinsic signal optical imaging"]
+
+    def __init__(self, folder_path: DirectoryPath):
+        super().__init__(folder_path=folder_path)
+
+    def get_metadata(self) -> DeepDict:
+        # Automatically retrieve as much metadata as possible from the source files available
+        metadata = super().get_metadata()
+
+        return metadata
+
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["IntrinsicSignalOpticalImaging"] = dict()
+        metadata_schema["properties"]["IntrinsicSignalOpticalImaging"]["properties"] = {
+            "Module": {
+                "properties": {
+                    "name": {"type": "string"},
+                    "description": {"type": "string"},
+                },
+            },
+            "Images": {
+                "properties": {
+                    "name": {"type": "string"},
+                    "description": {"type": "string"},
+                },
+            },
+            "RawImage": {
+                "properties": {
+                    "name": {"type": "string"},
+                    "description": {"type": "string"},
+                },
+            },
+            "ProcessedImage": {
+                "properties": {
+                    "name": {"type": "string"},
+                    "description": {"type": "string"},
+                },
+            },
+        }
+        return metadata_schema
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict):
+        # Read Data
+        folder_path = self.source_data["folder_path"]
+        raw_image_path = folder_path / "BloodvesselPattern.tiff"
+        processed_image_path = folder_path / "IOS_imageOverlaidFinal.jpg"
+        with Image.open(raw_image_path) as image:
+            raw_image_array = np.array(image)
+        with Image.open(processed_image_path) as image:
+            processed_image_array = np.array(image)
+
+        # Add Data to NWBFile
+        isoi_metadata = metadata["IntrinsicSignalOpticalImaging"]
+        isoi_module = nwb_helpers.get_module(
+            nwbfile=nwbfile,
+            name=isoi_metadata["Module"]["name"],
+            description=isoi_metadata["Module"]["description"],
+        )
+        raw_image = GrayscaleImage(
+            name=isoi_metadata["RawImage"]["name"],
+            data=raw_image_array,
+            description=isoi_metadata["RawImage"]["description"],
+        )
+        processed_image = RGBImage(
```

Comment:
For the color case, did you check that the images is indeed read as RBG and in the right orientation? (just display after writing it or with the neurosfit. I have gotten some surprises in that direction with

Reply:
Yes, I checked this earlier it matches the image coming out of the tiff.

Interestingly, when I show the image using PIL it's flipped left/right relative to plt.imshow(np.array(img))...

Reply:
As long as it is in the expected orientation when someone reads it from nwb that should be fine.

Reply:
Although the overlaid image is flipped left/right relative to the original raw_image...will make a note and ask.